***1er CASO***

***FIBRA DE CARBONO 2***

In [50]:
from sympy import *

# Datos Generales (CARBONO 2)

b = .3  # m
h = .55  # m
hef = h - 0.08  # m
dsup = 0.08  # m

# Concreto
fc = 210  # kg/cm2
Ec = 15100 * (fc) ** .5  # kg/cm2
ecu = 0.003  # Deformacion ultima del concreto
fy = 4200  # kg/cm2
Es = 2 * 10 ** 6  # kg/cm2
esu = 0.0021  # Deformacion ultima del acero
nrm = Es / Ec
# FRP : Son brindados por el fabricante Ojo !
ffu = 9860  # kg/cm2 : Tension de ruptura del FRP
efu = 0.01  # Deformacion de ruptura del FRP
Efrp = 958000  # kgf/cm2 Modulo de elasticidad del FRP
tf = .001  # m : Espesor de la lamina de FRP
nrmfrp = Efrp / Ec

Mwdl = [-2.75]  # 1er Paño : Carga Muerta
Mcs = [-6.83]  # 1er Paño : Cargas de Servicio

n = 3
diam = 5 / 8  # pulg
Adiam1 = pi * (diam * 2.54) ** 2 / 4  # cm2
As11 = (n * Adiam1) * 10 ** -4  # m2
n = 0
diam = 3 / 4  # pulg
As12 = (n * Adiam1) * 10 ** -4  # m2
rho1 = As11 / (b * hef)  # Cuantia de Refuerzo
w = rho1 * fy / fc
phi = 0.9  # Coeficiente de Reduccion para Flexion
phiMn1 = phi * (b * hef ** 2 * w * fc * 10 * (1 - 0.59 * w))  # tnf-m
print("phiMn1=", round(phiMn1, 2), "tnf-m")

CE = 0.85
ffuef = CE * ffu
efuef = CE * efu
print("ffuef=", round(ffuef, 2), "kg/cm2")
print("efuef=", round(efuef, 2))

beta1 = 0.85
rhob = 0.85 * fc * beta1 / fy * (6117 / (fy + 6117))  # Segun Teodoro Harmsen es la Cuantia Balanceada
print("rhob=", round(rhob, 5))

nf = 1  # Numero de Capas de FRP
bf = .25  # m : Ancho de la Lamina de FRP
Af = nf * bf * tf  # m2 : Area de FRP
rhof = Af / (b * hef)  # Cuantia de FRP
print('Af =', round(Af, 4), 'm2')
print('rhof =', round(rhof, 4))

k = -rho1 * nrm + sqrt((rho1 * nrm) ** 2 + 2 * rho1 * nrm)
Icr = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2
print('k =', round(k, 2))
print("Icr=", round(Icr, 5), "m4")

df = h + tf / 2  # m : Distancia al centroide de la fibra
ebi = -Mwdl[0] * (df - k * hef) / (Icr * Ec * 10)  # m : Deformacion por Flexion
print("ebi=", round(ebi, 5))

efd = .41 * (fc / (nrm * Efrp * tf * 100)) ** .5
if efd < 0.9 * efuef:
    print('efd < 0.9*efuef', '::', round(efd, 5), '<', round(0.9 * efuef, 5))
    print('OK')
else:
    print('efd > 0.9*efuef', '::', round(efd, 5), '>', round(0.9 * efuef, 5))
    print('NO OK')

c = 0.098778  # m : Distancia del eje neutro

# Suponemos falla del concreto, entonces ecu=0.003
efe = ecu * (df - c) / c - ebi
if efe > efd:
    print('efe > efd', '::', round(efe, 5), '>', round(efd, 5))
    print('El concreto aun no ha fallado')
    efe = efd
    ec = (efe + ebi) * c / (df - c)
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
else:
    print('efe < efd', '::', round(efe, 5), '<', round(efd, 5))
    print('El concreto ha fallado')
    ec = ecu
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
es = (efe + ebi) * (hef - c) / (df - c)
if es > esu:
    print('es > esu', '::', round(es, 5), '>', round(esu, 5))
    print('El acero ha fallado')
    print('fs=fy')
    fs = fy * 10
ffe = Efrp * 10 * efe
if ffe > ffuef * 10:
    print('ffe > ffuef', '::', round(ffe, 2), '>', round(ffuef * 10, 2))
    print('El FRP ha fallado')
    print('ffe=ffuef')
    ffe = ffuef
else:
    print('ffe < ffuef', '::', round(ffe, 2), '<', round(ffuef * 10, 2))
    print('El FRP aun no ha fallado')
ecef = 1.7 * fc / Ec
beta1 = (4 * ecef - ec) / (6 * ecef - 2 * ec)
alpha1 = (3 * ecef * ec - ec ** 2) / (3 * beta1 * ecef ** 2)
print('ecef =', round(ecef, 5))
print('beta1 =', round(beta1, 4))
print('alpha1 =', round(alpha1, 4))
cc = alpha1 * fc * 10 * beta1 * c * b
Ts = As11 * fs
Tf = Af * ffe
print('cc =', round(cc, 2), 'tnf')
print('Ts =', round(Ts, 2), 'tnf')
print('Tf =', round(Tf, 2), 'tnf')
print('Error =', round(cc - (Ts + Tf), 5), 'tnf')

print('c =', round(c, 8))

Mns = As11 * fs * (hef - beta1 * c / 2)
Mnf = Af * ffe * (df - beta1 * c / 2)

print('Mns =', round(Mns, 2), 'tnf-m')
print('Mnf =', round(Mnf, 2), 'tnf-m')

if es > 0.005:
    phi = 0.9
else:
    phi = 0.65 + 0.25 * (es - esu) / (0.005 - esu)
sif = 0.85
phiMn = phi * (Mns + sif * Mnf)
print('phiMn =', round(phiMn, 2), 'tnf-m')

Icrn = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2 + nrm * As12 * (k * hef - dsup) ** 2 + nrmfrp * Af * (df - k * hef) ** 2
print('Icrn =', round(Icrn, 5), 'm4')

fss = nrm * -Mcs[0] * (hef - k * hef) / Icrn
ffs = nrmfrp * -Mcs[0] * (df - k * hef) / Icrn - ebi * Efrp * 10
print('fss =', round(fss, 2), 'tnf/m2')
print('ffs =', round(ffs, 2), 'tnf/m2')

if fss / (fy * 10) < .8:
    print('fss/fy < 0.8', '::', round(fss / (fy * 10), 2), '<', .8)
    print('OK')
else:
    print('fss/fy > 0.8', '::', round(fss / (fy * 10), 2), '>', .8)
    print('NO OK')

if ffs / (ffuef * 10) < .55:
    print('ffs/ffuef < 0.55', '::', round(ffs / (ffuef * 10), 2), '<', .55)
    print('OK')
else:
    print('ffs/ffuef > 0.55', '::', round(ffs / (ffuef * 10), 2), '>', .55)
    print('NO OK')

fcs = -Mcs[0] * k * hef / Icrn
print('fcs =', round(fcs, 2), 'tnf/m2')

if fcs / (fc * 10) < .45:
    print('fcs/fc < 0.45', '::', round(fcs / (fc * 10), 2), '<', .45)
    print('OK')
else:
    print('fcs/fc > 0.45', '::', round(fcs / (fc * 10), 2), '>', .45)
    print('NO OK')

print('----------------------------------------------------------------')

phiMn1= 10.03 tnf-m
ffuef= 8381.0 kg/cm2
efuef= 0.01
rhob= 0.02142
Af = 0.0003 m2
rhof = 0.0018
k = 0.24
Icr= 0.00084 m4
ebi= 0.00066
efd < 0.9*efuef :: 0.00635 < 0.00765
OK
efe > efd :: 0.01306 > 0.00635
El concreto aun no ha fallado
El valor ec = 0.00153 :: efe=efd
es > esu :: 0.00576 > 0.0021
El acero ha fallado
fs=fy
ffe < ffuef :: 60828.14 < 83810.0
El FRP aun no ha fallado
ecef = 0.00163
beta1 = 0.7426
alpha1 = 0.8687
cc = 40.15 tnf
Ts = 24.94 tnf
Tf = 15.21 tnf
Error = -0.00063 tnf
c = 0.098778
Mns = 10.81 tnf-m
Mnf = 7.81 tnf-m
phiMn = 15.70 tnf-m
Icrn = 0.00104 m4
fss = 21294.02 tnf/m2
ffs = 6210.75 tnf/m2
fss/fy < 0.8 :: 0.51 < 0.8
OK
ffs/ffuef < 0.55 :: 0.07 < 0.55
OK
fcs = 742.28 tnf/m2
fcs/fc < 0.45 :: 0.35 < 0.45
OK
----------------------------------------------------------------


***FIBRA DE VIDRIO***

In [2]:
from sympy import *

# Datos Generales (VIDRIO)

b = .3  # m
h = .5  # m
hef = h - 0.08  # m
dsup = 0.08  # m

# Concreto
fc = 280  # kg/cm2
Ec = 15100 * (fc) ** .5  # kg/cm2
ecu = 0.003  # Deformacion ultima del concreto
fy = 4200  # kg/cm2
Es = 2 * 10 ** 6  # kg/cm2
esu = 0.0021  # Deformacion ultima del acero
nrm = Es / Ec
# FRP : Son brindados por el fabricante Ojo !
ffu = 5750  # kg/cm2 : Tension de ruptura del FRP
efu = 0.022  # Deformacion de ruptura del FRP
Efrp = 26.1 * 10 ** 4  # kgf/cm2 Modulo de elasticidad del FRP
tf = .0013  # m : Espesor de la lamina de FRP
nrmfrp = Efrp / Ec

Mwdl = [-3.5]  # 1er Paño : Carga Muerta
Mcs = [-8.5]  # 1er Paño : Cargas de Servicio

n = 3
diam = 5 / 8  # pulg
Adiam1 = pi * (diam * 2.54) ** 2 / 4  # cm2
As11 = (n * Adiam1) * 10 ** -4  # m2
n = 0
diam = 3 / 4  # pulg
As12 = (n * Adiam1) * 10 ** -4  # m2
rho1 = As11 / (b * hef)  # Cuantia de Refuerzo
w = rho1 * fy / fc
phi = 0.9  # Coeficiente de Reduccion para Flexion
phiMn1 = phi * (b * hef ** 2 * w * fc * 10 * (1 - 0.59 * w))  # tnf-m
print("phiMn1=", round(phiMn1, 2), "tnf-m")

CE = 0.75
ffuef = CE * ffu
efuef = CE * efu
print("ffuef=", round(ffuef, 2), "kg/cm2")
print("efuef=", round(efuef, 2))

beta1 = 0.85
rhob = 0.85 * fc * beta1 / fy * (6117 / (fy + 6117))  # Segun Teodoro Harmsen es la Cuantia Balanceada
print("rhob=", round(rhob, 5))

nf = 2  # Numero de Capas de FRP
bf = .25  # m : Ancho de la Lamina de FRP
Af = nf * bf * tf  # m2 : Area de FRP
rhof = Af / (b * hef)  # Cuantia de FRP
print('Af =', round(Af, 4), 'm2')
print('rhof =', round(rhof, 4))

k = -rho1 * nrm + sqrt((rho1 * nrm) ** 2 + 2 * rho1 * nrm)
Icr = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2
print('k =', round(k, 2))
print("Icr=", round(Icr, 5), "m4")

df = h + tf / 2  # m : Distancia al centroide de la fibra
ebi = -Mwdl[0] * (df - k * hef) / (Icr * Ec * 10)  # m : Deformacion por Flexion
print("ebi=", round(ebi, 5))

efd = .41 * (fc / (nrm * Efrp * tf * 100)) ** .5
if efd < 0.9 * efuef:
    print('efd < 0.9*efuef', '::', round(efd, 5), '<', round(0.9 * efuef, 5))
    print('OK')
else:
    print('efd > 0.9*efuef', '::', round(efd, 5), '>', round(0.9 * efuef, 5))
    print('NO OK')

c = 0.0760103  # m : Distancia del eje neutro

# Suponemos falla del concreto, entonces ecu=0.003 ## Parece ser una suposicion Obligatoria !!
efe = ecu * (df - c) / c - ebi
if efe > efd:
    print('efe > efd', '::', round(efe, 5), '>', round(efd, 5))
    print('El concreto aun no ha fallado')
    efe = efd
    ec = (efe + ebi) * c / (df - c)
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
else:
    print('efe < efd', '::', round(efe, 5), '<', round(efd, 5))
    print('El concreto ha fallado')
    ec = ecu
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
es = (efe + ebi) * (hef - c) / (df - c)
if es > esu:
    print('es > esu', '::', round(es, 5), '>', round(esu, 5))
    print('El acero ha fallado')
    print('fs=fy')
    fs = fy * 10
else:
    print('es < esu', '::', round(es, 5), '<', round(esu, 5))
    print('El acero no ha fallado')
    print('fs=Es*es')
    fs = (Es * es) * 10
ffe = Efrp * 10 * efe
if ffe > ffuef * 10:
    print('ffe > ffuef', '::', round(ffe, 2), '>', round(ffuef * 10, 2))
    print('El FRP ha fallado')
    print('ffe=ffuef')
    ffe = ffuef
else:
    print('ffe < ffuef', '::', round(ffe, 2), '<', round(ffuef * 10, 2))
    print('El FRP aun no ha fallado')
ecef = 1.7 * fc / Ec
beta1 = (4 * ecef - ec) / (6 * ecef - 2 * ec)
alpha1 = (3 * ecef * ec - ec ** 2) / (3 * beta1 * ecef ** 2)
print('ecef =', round(ecef, 5))
print('beta1 =', round(beta1, 4))
print('alpha1 =', round(alpha1, 4))
cc = alpha1 * fc * 10 * beta1 * c * b
Ts = As11 * fs
Tf = Af * ffe
print('cc =', round(cc, 2), 'tnf')
print('Ts =', round(Ts, 2), 'tnf')
print('Tf =', round(Tf, 2), 'tnf')
print('Error =', round(cc - (Ts + Tf), 5), 'tnf')

print('c =', round(c, 8))

Mns = As11 * fs * (hef - beta1 * c / 2)
Mnf = Af * ffe * (df - beta1 * c / 2)

print('Mns =', round(Mns, 2), 'tnf-m')
print('Mnf =', round(Mnf, 2), 'tnf-m')

if es > 0.005:
    phi = 0.9
else:
    phi = 0.65 + 0.25 * (es - esu) / (0.005 - esu)
sif = 0.85
phiMn = phi * (Mns + sif * Mnf)
print('phiMn =', round(phiMn, 2), 'tnf-m')

Icrn = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2 + nrm * As12 * (k * hef - dsup) ** 2 + nrmfrp * Af * (
            df - k * hef) ** 2
print('Icrn =', round(Icrn, 5), 'm4')

fss = nrm * -Mcs[0] * (hef - k * hef) / Icrn
ffs = nrmfrp * -Mcs[0] * (df - k * hef) / Icrn - ebi * Efrp * 10
print('fss =', round(fss, 2), 'tnf/m2')
print('ffs =', round(ffs, 2), 'tnf/m2')

if fss / (fy * 10) < .8:
    print('fss/fy < 0.8', '::', round(fss / (fy * 10), 2), '<', .8)
    print('OK')
else:
    print('fss/fy > 0.8', '::', round(fss / (fy * 10), 2), '>', .8)
    print('NO OK')

if ffs / (ffuef * 10) < .55:
    print('ffs/ffuef < 0.55', '::', round(ffs / (ffuef * 10), 2), '<', .55)
    print('OK')
else:
    print('ffs/ffuef > 0.55', '::', round(ffs / (ffuef * 10), 2), '>', .55)
    print('NO OK')

fcs = -Mcs[0] * k * hef / Icrn
print('fcs =', round(fcs, 2), 'tnf/m2')

if fcs / (fc * 10) < .45:
    print('fcs/fc < 0.45', '::', round(fcs / (fc * 10), 2), '<', .45)
    print('OK')
else:
    print('fcs/fc > 0.45', '::', round(fcs / (fc * 10), 2), '>', .45)
    print('NO OK')

print('----------------------------------------------------------------')

phiMn1= 9.03 tnf-m
ffuef= 4312.5 kg/cm2
efuef= 0.02
rhob= 0.02856
Af = 0.0006 m2
rhof = 0.0052
k = 0.24
Icr= 0.00058 m4
ebi= 0.00095
efd < 0.9*efuef :: 0.01324 < 0.01485
OK
efe > efd :: 0.01581 > 0.01324
El concreto aun no ha fallado
El valor ec = 0.00254 :: efe=efd
es > esu :: 0.01150 > 0.0021
El acero ha fallado
fs=fy
ffe < ffuef :: 34552.1 < 43125.0
El FRP aun no ha fallado
ecef = 0.00188
beta1 = 0.8028
alpha1 = 0.9247
cc = 47.40 tnf
Ts = 24.94 tnf
Tf = 22.46 tnf
Error = -1.e-5 tnf
c = 0.0760103
Mns = 9.71 tnf-m
Mnf = 10.56 tnf-m
phiMn = 16.82 tnf-m
Icrn = 0.00069 m4
fss = 31236.23 tnf/m2
ffs = 2612.95 tnf/m2
fss/fy < 0.8 :: 0.74 < 0.8
OK
ffs/ffuef < 0.55 :: 0.06 < 0.55
OK
fcs = 1235.09 tnf/m2
fcs/fc < 0.45 :: 0.44 < 0.45
OK
----------------------------------------------------------------


***2do CASO***

***FIBRA DE CARBONO 2***

In [3]:
from sympy import *

# Datos Generales (CARBONO 2)

b = .3  # m
h = .5  # m
hef = h - 0.08  # m
dsup = 0.08  # m
Lab = 6  # m
Lbc = 8  # m

# Concreto
fc = 280  # kg/cm2
Ec = 15100 * (fc) ** .5  # kg/cm2
ecu = 0.003  # Deformacion ultima del concreto
fy = 4200  # kg/cm2
Es = 2 * 10 ** 6  # kg/cm2
esu = 0.0021  # Deformacion ultima del acero
nrm = Es / Ec
# FRP : Son brindados por el fabricante Ojo !
ffu = 31000  # kg/cm2 : Tension de ruptura del FRP
efu = 0.017  # Deformacion de ruptura del FRP
Efrp = 1650000  # kgf/cm2 Modulo de elasticidad del FRP
tf = .0012  # m : Espesor de la lamina de FRP
nrmfrp = Efrp / Ec

wd = .500  # kgf/m2 : Carga Muerta de la Viga
sco = .250  # kgf/m2 : Sobrecarga de uso de la Viga (Inicial)
scn = .650  # kgf/m2 : Sobrecarga de uso de la Viga (Nueva)
anch_t = 5  # m : Ancho Tributario

Mwdl = [-6]  # 1er Paño : Carga Muerta
Mcs = [-14]  # 1er Paño : Cargas de Servicio

n = 4
diam = 3 / 4  # pulg
Adiam1 = pi * (diam * 2.54) ** 2 / 4  # cm2
As11 = (n * Adiam1) * 10 ** -4  # m2
n = 0
diam = 3 / 4  # pulg
As12 = (n * Adiam1) * 10 ** -4  # m2
rho1 = As11 / (b * hef)  # Cuantia de Refuerzo
w = rho1 * fy / fc
phi = 0.9  # Coeficiente de Reduccion para Flexion
phiMn1 = phi * (b * hef ** 2 * w * fc * 10 * (1 - 0.59 * w))  # tnf-m
print("phiMn1=", round(phiMn1, 2), "tnf-m")

CE = 0.95
ffuef = CE * ffu
efuef = CE * efu
print("ffuef=", round(ffuef, 2), "kg/cm2")
print("efuef=", round(efuef, 2))

beta1 = 0.85
rhob = 0.85 * fc * beta1 / fy * (6117 / (fy + 6117))  # Segun Teodoro Harmsen es la Cuantia Balanceada
print("rhob=", round(rhob, 5))

nf = 2  # Numero de Capas de FRP
bf = .25  # m : Ancho de la Lamina de FRP
Af = nf * bf * tf  # m2 : Area de FRP
rhof = Af / (b * hef)  # Cuantia de FRP
print('Af =', round(Af, 4), 'm2')
print('rhof =', round(rhof, 4))

k = -rho1 * nrm + sqrt((rho1 * nrm) ** 2 + 2 * rho1 * nrm)
Icr = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2
print('k =', round(k, 2))
print("Icr=", round(Icr, 5), "m4")

df = h + tf / 2  # m : Distancia al centroide de la fibra
ebi = -Mwdl[0] * (df - k * hef) / (Icr * Ec * 10)  # m : Deformacion por Flexion
print("ebi=", round(ebi, 5))

efd = .41 * (fc / (nrm * Efrp * tf * 100)) ** .5
if efd < 0.9 * efuef:
    print('efd < 0.9*efuef', '::', round(efd, 5), '<', round(0.9 * efuef, 5))
    print('OK')
else:
    print('efd > 0.9*efuef', '::', round(efd, 5), '>', round(0.9 * efuef, 5))
    print('NO OK')

c = 0.1615195 # m : Distancia del eje neutro

# Suponemos falla del concreto, entonces ecu=0.003
efe = ecu * (df - c) / c - ebi
if efe > efd:
    print('efe > efd', '::', round(efe, 5), '>', round(efd, 5))
    print('El concreto aun no ha fallado')
    efe = efd
    ec = (efe + ebi) * c / (df - c)
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
else:
    print('efe < efd', '::', round(efe, 5), '<', round(efd, 5))
    print('El concreto ha fallado')
    ec = ecu
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
es = (efe + ebi) * (hef - c) / (df - c)
if es > esu:
    print('es > esu', '::', round(es, 5), '>', round(esu, 5))
    print('El acero ha fallado')
    print('fs=fy')
    fs = fy * 10
ffe = Efrp * 10 * efe
if ffe > ffuef * 10:
    print('ffe > ffuef', '::', round(ffe, 2), '>', round(ffuef * 10, 2))
    print('El FRP ha fallado')
    print('ffe=ffuef')
    ffe = ffuef
else:
    print('ffe < ffuef', '::', round(ffe, 2), '<', round(ffuef * 10, 2))
    print('El FRP aun no ha fallado')
ecef = 1.7 * fc / Ec
beta1 = (4 * ecef - ec) / (6 * ecef - 2 * ec)
alpha1 = (3 * ecef * ec - ec ** 2) / (3 * beta1 * ecef ** 2)
print('ecef =', round(ecef, 5))
print('beta1 =', round(beta1, 4))
print('alpha1 =', round(alpha1, 4))
cc = alpha1 * fc * 10 * beta1 * c * b
Ts = As11 * fs
Tf = Af * ffe
print('cc =', round(cc, 2), 'tnf')
print('Ts =', round(Ts, 2), 'tnf')
print('Tf =', round(Tf, 2), 'tnf')
print('Error =', round(cc - (Ts + Tf), 5), 'tnf')

print('c =', round(c, 8))

Mns = As11 * fs * (hef - beta1 * c / 2)
Mnf = Af * ffe * (df - beta1 * c / 2)

print('Mns =', round(Mns, 2), 'tnf-m')
print('Mnf =', round(Mnf, 2), 'tnf-m')

if es > 0.005:
    phi = 0.9
else:
    phi = 0.65 + 0.25 * (es - esu) / (0.005 - esu)
sif = 0.85
phiMn = phi * (Mns + sif * Mnf)
print('phiMn =', round(phiMn, 2), 'tnf-m')

Icrn = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2 + nrm * As12 * (k * hef - dsup) ** 2 + nrmfrp * Af * (
            df - k * hef) ** 2
print('Icrn =', round(Icrn, 5), 'm4')

fss = nrm * -Mcs[0] * (hef - k * hef) / Icrn
ffs = nrmfrp * -Mcs[0] * (df - k * hef) / Icrn - ebi * Efrp * 10
print('fss =', round(fss, 2), 'tnf/cm2')
print('ffs =', round(ffs, 2), 'tnf/cm2')

if fss / (fy * 10) < .8:
    print('fss/fy < 0.8', '::', round(fss / (fy * 10), 2), '<', .8)
    print('OK')
else:
    print('fss/fy > 0.8', '::', round(fss / (fy * 10), 2), '>', .8)
    print('NO OK')

if ffs / (ffuef * 10) < .55:
    print('ffs/ffuef < 0.55', '::', round(ffs / (ffuef * 10), 2), '<', .55)
    print('OK')
else:
    print('ffs/ffuef > 0.55', '::', round(ffs / (ffuef * 10), 2), '>', .55)
    print('NO OK')

fcs = -Mcs[0] * k * hef / Icrn
print('fcs =', round(fcs, 2), 'tnf/cm2')

if fcs / (fc * 10) < .45:
    print('fcs/fc < 0.45', '::', round(fcs / (fc * 10), 2), '<', .45)
    print('OK')
else:
    print('fcs/fc > 0.45', '::', round(fcs / (fc * 10), 2), '>', .45)
    print('NO OK')

print('----------------------------------------------------------------')

phiMn1= 16.65 tnf-m
ffuef= 29450.0 kg/cm2
efuef= 0.02
rhob= 0.02856
Af = 0.0006 m2
rhof = 0.0048
k = 0.31
Icr= 0.00098 m4
ebi= 0.00090
efd < 0.9*efuef :: 0.00548 < 0.01454
OK
efe < efd :: 0.00540 < 0.00548
El concreto ha fallado
El valor ec = 0.003 :: efe=efd
es > esu :: 0.00480 > 0.0021
El acero ha fallado
fs=fy
ffe < ffuef :: 89144.61 < 294500.0
El FRP aun no ha fallado
ecef = 0.00188
beta1 = 0.8552
alpha1 = 0.8736
cc = 101.37 tnf
Ts = 47.88 tnf
Tf = 53.49 tnf
Error = -2.e-5 tnf
c = 0.1615195
Mns = 16.80 tnf-m
Mnf = 23.08 tnf-m
phiMn = 32.16 tnf-m
Icrn = 0.00151 m4
fss = 21133.81 tnf/cm2
ffs = 7538.26 tnf/cm2
fss/fy < 0.8 :: 0.50 < 0.8
OK
ffs/ffuef < 0.55 :: 0.03 < 0.55
OK
fcs = 1219.67 tnf/cm2
fcs/fc < 0.45 :: 0.44 < 0.45
OK
----------------------------------------------------------------


***FIBRA DE VIDRIO***

In [4]:
from sympy import *

# Datos Generales (VIDRIO)

b = .3  # m
h = .5  # m
hef = h - 0.08  # m
dsup = 0.08  # m

# Concreto
fc = 280  # kg/cm2
Ec = 15100 * (fc) ** .5  # kg/cm2
ecu = 0.003  # Deformacion ultima del concreto
fy = 4200  # kg/cm2
Es = 2 * 10 ** 6  # kg/cm2
esu = 0.0021  # Deformacion ultima del acero
nrm = Es / Ec
# FRP : Son brindados por el fabricante Ojo !
ffu = 5750  # kg/cm2 : Tension de ruptura del FRP
efu = 0.022  # Deformacion de ruptura del FRP
Efrp = 26.1 * 10 ** 4  # kgf/cm2 Modulo de elasticidad del FRP
tf = .0013  # m : Espesor de la lamina de FRP
nrmfrp = Efrp / Ec

Mwdl = [-6]  # 1er Paño : Carga Muerta
Mcs = [-14]  # 1er Paño : Cargas de Servicio

n = 4
diam = 3 / 4  # pulg
Adiam1 = pi * (diam * 2.54) ** 2 / 4  # cm2
As11 = (n * Adiam1) * 10 ** -4  # m2
n = 0
diam = 3 / 4  # pulg
As12 = (n * Adiam1) * 10 ** -4  # m2
rho1 = As11 / (b * hef)  # Cuantia de Refuerzo
w = rho1 * fy / fc
phi = 0.9  # Coeficiente de Reduccion para Flexion
phiMn1 = phi * (b * hef ** 2 * w * fc * 10 * (1 - 0.59 * w))  # tnf-m
print("phiMn1=", round(phiMn1, 2), "tnf-m")

CE = 0.75
ffuef = CE * ffu
efuef = CE * efu
print("ffuef=", round(ffuef, 2), "kg/cm2")
print("efuef=", round(efuef, 2))

beta1 = 0.85
rhob = 0.85 * fc * beta1 / fy * (6117 / (fy + 6117))  # Segun Teodoro Harmsen es la Cuantia Balanceada
print("rhob=", round(rhob, 5))

nf = 12 # Numero de Capas de FRP
bf = .25  # m : Ancho de la Lamina de FRP
Af = nf * bf * tf  # m2 : Area de FRP
rhof = Af / (b * hef)  # Cuantia de FRP
print('Af =', round(Af, 4), 'm2')
print('rhof =', round(rhof, 4))

k = -rho1 * nrm + sqrt((rho1 * nrm) ** 2 + 2 * rho1 * nrm)
Icr = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2
print('k =', round(k, 2))
print("Icr=", round(Icr, 5), "m4")

df = h + tf / 2  # m : Distancia al centroide de la fibra
ebi = -Mwdl[0] * (df - k * hef) / (Icr * Ec * 10)  # m : Deformacion por Flexion
print("ebi=", round(ebi, 5))

efd = .41 * (fc / (nrm * Efrp * tf * 100)) ** .5
if efd < 0.9 * efuef:
    print('efd < 0.9*efuef', '::', round(efd, 5), '<', round(0.9 * efuef, 5))
    print('OK')
else:
    print('efd > 0.9*efuef', '::', round(efd, 5), '>', round(0.9 * efuef, 5))
    print('NO OK')

c = 0.162773  # m : Distancia del eje neutro

# Suponemos falla del concreto, entonces ecu=0.003 ## Parece ser una suposicion Obligatoria !!
efe = ecu * (df - c) / c - ebi
if efe > efd:
    print('efe > efd', '::', round(efe, 5), '>', round(efd, 5))
    print('El concreto aun no ha fallado')
    efe = efd
    ec = (efe + ebi) * c / (df - c)
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
else:
    print('efe < efd', '::', round(efe, 5), '<', round(efd, 5))
    print('El concreto ha fallado')
    ec = ecu
    print('El valor ec =', round(ec, 5), '::', 'efe=efd')
es = (efe + ebi) * (hef - c) / (df - c)
if es > esu:
    print('es > esu', '::', round(es, 5), '>', round(esu, 5))
    print('El acero ha fallado')
    print('fs=fy')
    fs = fy * 10
else:
    print('es < esu', '::', round(es, 5), '<', round(esu, 5))
    print('El acero no ha fallado')
    print('fs=Es*es')
    fs = (Es * es) * 10
ffe = Efrp * 10 * efe
if ffe > ffuef * 10:
    print('ffe > ffuef', '::', round(ffe, 2), '>', round(ffuef * 10, 2))
    print('El FRP ha fallado')
    print('ffe=ffuef')
    ffe = ffuef
else:
    print('ffe < ffuef', '::', round(ffe, 2), '<', round(ffuef * 10, 2))
    print('El FRP aun no ha fallado')
ecef = 1.7 * fc / Ec
beta1 = (4 * ecef - ec) / (6 * ecef - 2 * ec)
alpha1 = (3 * ecef * ec - ec ** 2) / (3 * beta1 * ecef ** 2)
print('ecef =', round(ecef, 5))
print('beta1 =', round(beta1, 4))
print('alpha1 =', round(alpha1, 4))
cc = alpha1 * fc * 10 * beta1 * c * b
Ts = As11 * fs
Tf = Af * ffe
print('cc =', round(cc, 2), 'tnf')
print('Ts =', round(Ts, 2), 'tnf')
print('Tf =', round(Tf, 2), 'tnf')
print('Error =', round(cc - (Ts + Tf), 5), 'tnf')

print('c =', round(c, 8))

Mns = As11 * fs * (hef - beta1 * c / 2)
Mnf = Af * ffe * (df - beta1 * c / 2)

print('Mns =', round(Mns, 2), 'tnf-m')
print('Mnf =', round(Mnf, 2), 'tnf-m')

if es > 0.005:
    phi = 0.9
else:
    phi = 0.65 + 0.25 * (es - esu) / (0.005 - esu)
sif = 0.85
phiMn = phi * (Mns + sif * Mnf)
print('phiMn =', round(phiMn, 2), 'tnf-m')

Icrn = b * (k * hef) ** 3 / 3 + nrm * As11 * (hef - k * hef) ** 2 + nrm * As12 * (k * hef - dsup) ** 2 + nrmfrp * Af * (df - k * hef)**2
print('Icrn =', round(Icrn, 5), 'm4')

fss = nrm * -Mcs[0] * (hef - k * hef) / Icrn
ffs = nrmfrp * -Mcs[0] * (df - k * hef) / Icrn - ebi * Efrp * 10
print('fss =', round(fss, 2), 'tnf/m2')
print('ffs =', round(ffs, 2), 'tnf/m2')

if fss / (fy * 10) < .8:
    print('fss/fy < 0.8', '::', round(fss / (fy * 10), 2), '<', .8)
    print('OK')
else:
    print('fss/fy > 0.8', '::', round(fss / (fy * 10), 2), '>', .8)
    print('NO OK')

if ffs / (ffuef * 10) < .55:
    print('ffs/ffuef < 0.55', '::', round(ffs / (ffuef * 10), 2), '<', .55)
    print('OK')
else:
    print('ffs/ffuef > 0.55', '::', round(ffs / (ffuef * 10), 2), '>', .55)
    print('NO OK')

fcs = -Mcs[0] * k * hef / Icrn
print('fcs =', round(fcs, 2), 'tnf/m2')

if fcs / (fc * 10) < .45:
    print('fcs/fc < 0.45', '::', round(fcs / (fc * 10), 2), '<', .45)
    print('OK')
else:
    print('fcs/fc > 0.45', '::', round(fcs / (fc * 10), 2), '>', .45)
    print('NO OK')

print('----------------------------------------------------------------')

phiMn1= 16.65 tnf-m
ffuef= 4312.5 kg/cm2
efuef= 0.02
rhob= 0.02856
Af = 0.0039 m2
rhof = 0.031
k = 0.31
Icr= 0.00098 m4
ebi= 0.00090
efd < 0.9*efuef :: 0.01324 < 0.01485
OK
efe < efd :: 0.00533 < 0.01324
El concreto ha fallado
El valor ec = 0.003 :: efe=efd
es > esu :: 0.00474 > 0.0021
El acero ha fallado
fs=fy
ffe < ffuef :: 13916.26 < 43125.0
El FRP aun no ha fallado
ecef = 0.00188
beta1 = 0.8552
alpha1 = 0.8736
cc = 102.16 tnf
Ts = 47.88 tnf
Tf = 54.27 tnf
Error = 3.e-5 tnf
c = 0.162773
Mns = 16.78 tnf-m
Mnf = 23.39 tnf-m
phiMn = 32.18 tnf-m
Icrn = 0.00153 m4
fss = 20923.77 tnf/m2
ffs = 1157.50 tnf/m2
fss/fy < 0.8 :: 0.50 < 0.8
OK
ffs/ffuef < 0.55 :: 0.03 < 0.55
OK
fcs = 1207.55 tnf/m2
fcs/fc < 0.45 :: 0.43 < 0.45
OK
----------------------------------------------------------------


***COMPARACIONES***

|Fibra|CE|nf|tf(m)|bf(m)|Af(m2)|c| $\varepsilon _c$|$\varepsilon _{fe}$|$\varepsilon _s$
|-|-|-|-|-|-|-|-|-|-|
**Caso 1**
|Carbono 1|0.95|1|0.0012|0.3|0.00036|0.086617|0.0017|0.0071|0.008|
|Carbono 2|0.95|1|0.0012|0.25|0.0003|0.10089|0.0016|0.0054|0.0051|
|Vidrio|0.75|2|0.0013|0.25|0.00065|0.0760103|0.0025|0.013|0.012|
**Caso 2**
|Carbono 1|0.95|2|0.0012|0.3|0.00072|0.210792|0.0038|0.0044|0.0052|
|Carbono 2|0.95|2|0.0012|0.25|0.0006|0.1615195|0.003|0.0054|0.0048|
|Vidrio|0.75|12|0.0013|0.25|0.0039|0.162773|0.003|0.0053|0.0047|

Comentarios:
- Como se puede revisar en la tabla claramente emplear fibras de carbono resulta mas conveniente, ya que no solo ofrece seguridad sino tambien un menor precio para su colocacion despues de un analisis y diseño.
- El caso 2 para la fibra de vidrio no es factible debido al numero de capas que se necesitan (12) para poder satisfacer las condiciones de diseño.
- Se puede rescatar el hecho de que la fibra en todos los casos no alcanza su deformacion ultima, pero los demas componentes como el acero y el concreto si lo hacen. Importante !
- Cuanto mayor es la nueva demanda mayores son los requerimientos para la fibra, para los 2 casos de carbono el incremento de carga significo la adicion de una nueva capa.